src/herder/HerderSCPDriver.cpp

```diff
@@ -711,22 +723,82 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
                 {
                     highestTxSet = cTxSet;
                     highest = sv.txSetHash;
+                    if (candidateCloseTimes.size() == 0)
+                    {
+                        throw std::logic_error(
+                            "Missing candidate closeTimes list");
+                    }
+                    highestTxSetIndex =
+                        make_optional<size_t>(candidateCloseTimes.size() - 1);
+                    if (candidateCloseTimes[*highestTxSetIndex] != sv.closeTime)
+                    {
+                        throw std::logic_error("Incorrect highestTxSet index "
+                                               "into candidate closeTimes");
+                    }
+                    if (svSigned)
+                    {
+                        if (sv.ext.v() != STELLAR_VALUE_SIGNED)
+                        {
+                            throw std::logic_error(
+                                "Unsigned candidate StellarValue");
+                        }
+                        highestTxSetSignature =
+                            make_optional<LedgerCloseValueSignature>(
+                                sv.ext.lcValueSignature());
+                    }
                 }
             }
         }
+
+        if (!highestTxSet)
+        {
+            throw std::logic_error("No highest txSet");
+        }
         // make a copy as we're about to modify it and we don't want to mess
         // with the txSet cache
         bestTxSet = std::make_shared<TxSetFrame>(*highestTxSet);
     }
 
+    std::vector<TransactionFrameBasePtr> removed;
+    if (!highestTxSetIndex)
+    {
```

Comment:
same thing: this is just writing code that attempts to verify itself for correctness - we just end up with confusing code

Reply:
Alright, removed.